Support omit timer from NetworkState

cathay4t opened the pull request:
In kubernetes-nmstate, the output of `nmstatectl show` will be used to
track the changes of network. The changes of DHCP lift-time (e.g.
`preferred-life-time` and `valid-life-time`) generates false
notification on network changes.

Introduced `NetworkState.omit_timer()` and `nmstatectl show
--omit-timer` to remove these count down timer from output.

With `preferred-life-time` and `valid-life-time` properties omitted,
this patch also introduced `dynamic` property to ip address section
helping nmstate and user determine which IP address is coming from
DHCP/IPv6-RA.

The linux bridge `gc-timer` and `hello-timer` will also be purged by
`NetworkState.omit_timer()`. call.

Integration test cases included.

Resolves: https://issues.redhat.com/browse/RHEL-131158

jvaclav-rh commented on tests/integration/dynamic_ip_test.py:
```diff
@@ -274,6 +275,7 @@ def test_static_ip_with_auto_ip_enabled(dhcpcli_up):
             IPV4_ADDRESS4, 24, valid_lft="30sec", prefferred_lft="30sec"
         ),
     ]
+    print("HAHA ", ipv4_state)
```
print() from debugging

cathay4t replied:
Removed.

jvaclav-rh commented on rust/src/lib/ip.rs:
It might be worth mentioning the changes done by L346-348 in this comment section too.

cathay4t replied:
We already has `* Remove auto IP address.`.

cathay4t committed:
Support purging timer from NetworkState

In kubernetes-nmstate, the output of `nmstatectl show` will be used to
track the changes of network. The changes of DHCP lift-time (e.g.
`preferred-life-time` and `valid-life-time`) generates false
notification on network changes.

Introduced `NetworkState.purge_timer()` and `nmstatectl show
--purge-timer` to remove these count down timer from output.

With `preferred-life-time` and `valid-life-time` properties purged,
this patch also introduced `dynamic: true` property to ip address
section helping nmstate and user determine which IP address is coming
from DHCP/IPv6-RA. The `dynamic` property will be set to None for static
IP address when querying.

The linux bridge `gc-timer` and `hello-timer` will also be purged by
`NetworkState.purge_timer()`. call.

Integration test cases included.

Resolves: https://issues.redhat.com/browse/RHEL-131158

Signed-off-by: Gris Ge <fge@redhat.com>

cathay4t committed:
test: Fix `test_ipv4_dhcp_switch_on_to_off`

The in commit 4d0f0829, we will treat DHCP off along with empty IP
address as stack disabled.

Even the apply and verification action passed, but the test code
`assertlib.assert_state()` does not handle this properly, this causing
test case `test_switch_auto_to_static_with_dynamic_ips` fails
complaining about `enabled` has unexpected `False`.

Fixed by add a static IP as the description said.

Signed-off-by: Gris Ge <fge@redhat.com>

cathay4t commented:
Closing as OCP already has code purging IP lifetime, let's not risk IP stack code changes.
